sql/plan/show_create_table.go

```diff
@@ -184,13 +176,70 @@ func produceCreateTableStatement(table sql.Table) string {
 		colStmts = append(colStmts, primaryKey)
 	}
 
+	for _, index := range i.indexes {
+		// The primary key may or may not be declared as an index by the table. Don't print it twice if it's here.
+		if isPrimaryKeyIndex(index, table) {
+			continue
+		}
+
+		var indexCols []string
+		for _, expr := range index.Expressions() {
+			col := getColumnFromIndexExpr(expr, table)
+			if col != nil {
+				indexCols = append(indexCols, fmt.Sprintf("`%s`", col.Name))
+			}
+		}
+
+		unique := ""
+		if index.IsUnique() {
+			unique = "UNIQUE "
+		}
+
+		key := fmt.Sprintf("  %sKEY `%s` (%s)", unique, index.ID(), strings.Join(indexCols, ","))
+		colStmts = append(colStmts, key)
+	}
+
 	return fmt.Sprintf(
 		"CREATE TABLE `%s` (\n%s\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4",
 		table.Name(),
 		strings.Join(colStmts, ",\n"),
 	)
 }
 
+// isPrimaryKeyIndex returns whether the index given matches the table's primary key columns. Order is not considered.
```

Comment:
I think order is too important not to be considered. Perhaps add a TODO?

Reply:
Dolt itself can't support this yet because we don't store the order of primary key columns, which is why I didn't fix it yet. But yes, I'll add a TODO for this.